app/services/invoices/refresh_draft_service.rb

```diff
@@ -17,24 +18,39 @@ def call
       return result unless invoice.draft?
 
       ActiveRecord::Base.transaction do
-        invoice.credit_notes.destroy_all
-        invoice.credits.destroy_all
-        invoice.wallet_transactions.destroy_all
+        cn_subscription_ids = invoice.credit_notes.map do |cn|
+          { credit_note_id: cn.id, subscription_id: cn.fees.pick(:subscription_id) }
+        end
+        invoice.credit_notes.each { |cn| cn.items.update_all(fee_id: nil) }
+
         invoice.fees.destroy_all
         invoice.invoice_subscriptions.destroy_all
-
         invoice.update!(vat_rate: invoice.customer.applicable_vat_rate)
 
-        Invoices::CalculateFeesService.call(
+        calculate_result = Invoices::CalculateFeesService.call(
           invoice:,
           subscriptions: Subscription.find(subscription_ids),
           timestamp: invoice.created_at.to_i,
+          from_action:,
         )
+
+        invoice.credit_notes.each do |credit_note|
+          subscription_id = cn_subscription_ids.find { |h| h[:credit_note_id] == credit_note.id }[:subscription_id]
+          fee = invoice.fees.subscription.find_by(subscription_id:)
+          credit_note.items.update_all(fee_id: fee.id)
```

Comment:
Rails/SkipsModelValidations: Avoid using `update_all` because it skips validations. (https://guides.rubyonrails.org/active_record_validations.html#skipping-validations)